wandb/sdk/wandb_init.py

```diff
@@ -625,6 +629,11 @@ def init(self) -> Union[Run, RunDisabled, None]:  # noqa: C901
 
         self._wl._global_run_stack.append(run)
         self.run = run
+
+        # put artifacts in run config
+        for k, v in self.init_artifact_config.items():
```

Comment:
Is there a reason why this is done after the Run is created?  (im guessing there is.. i just havent been able to mentally follow the flow in this case)

Reply:
Yeah, since putting artifacts in the config automatically calls `use_artifact` we need the run to be upserted for those calls to not fail

Reply:
Got it. thanks.